core/headerchain.go

```diff
@@ -1246,6 +1247,33 @@ func (hc *HeaderChain) ComputeEfficiencyScore(parent *types.WorkObject) (uint16,
 	return ewma, nil
 }
 
+// CalcMaxBaseFee takes an average of the base fee over past 100 blocks
+func (hc *HeaderChain) CalcMaxBaseFee(block *types.WorkObject) *big.Int {
+	// Max Base fee can only be calculated in zone
+	if hc.NodeCtx() != common.ZONE_CTX {
+		return nil
+	}
+	// get the parent block
+	parent := hc.GetBlockByHash(block.ParentHash(hc.NodeCtx()))
+	if parent == nil {
+		return nil
```

Comment:
I would recommend returning an error here as this is very bad

Reply:
Fixed